bin/plugin/commands/performance.js

```diff
@@ -323,14 +323,14 @@ async function runPerformanceTests( branches, options ) {
 			log( `        >> Fetching the ${ fancyBranch } branch` );
 			// @ts-ignore
 			await SimpleGit( buildPath ).reset( 'hard' ).checkout( branch );
-
-			log( `        >> Building the ${ fancyBranch } branch` );
-			await runShellScript(
-				'npm ci && npm run prebuild:packages && node ./bin/packages/build.js && npx wp-scripts build',
-				buildPath
-			);
 		}
 
+		log( `        >> Building the ${ fancyBranch } branch` );
+		await runShellScript(
+			'npm ci && npm run prebuild:packages && node ./bin/packages/build.js && npx wp-scripts build',
+			buildPath
+		);
+
```

Comment:
In https://github.com/WordPress/gutenberg/pull/45737 we've introduced a potential optimization to the performance job, the problem is that it resulted in the job comparing the wrong thing.

Basically if a branch was equivalent to the "test branch", we reused the same code to gain some time and avoid the git clone, the issue is that we didn't build the code in that case, meaning we were using the WP trunk version of Gutenberg and not the actual branch.

(Currently WP trunk's version does have the `'/navigation/single'` support in the site editor which explains why this random change fixed the perf job.

The bummer is that you can say that for the last three months, the perf jobs in PRs were useless. Fortunately for us, the perf job in trunk was correct and the numbers reported on code health are correct.

Reply:
I guess there's an argument to be made that the perf job is more important in trunk than PRs and in which case, we could remove it from PRs. I was resistant to this idea but this is proof that we could consider removing it from PRs. But let's keep this discussion separate. Also, if we do really remove it from PRs, I'd like to find a way to enable it using a label or something for PRs that actually explicitly need performance.

Reply:
That also explains why I was completely unable to repro locally. 💥 
Nice catch, though the bug is indeed a nasty one! 🐛

Reply:
/cc @kevin940726 @t-hamano